### PR TITLE
fix(GiniCaptureSDK): trigger error screen on upload failure

### PR DESCRIPTION
**Pull request by vizaknaidavid**
PIA-3454

[PIA-3454]: https://ginis.atlassian.net/browse/PIA-3454?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift [hunk 1]
```diff
@@ -51,7 +51,6 @@ public final class DocumentService: DocumentServiceProtocol {
                     guard errorOccurred == false else {
                         return
                     }
-                    errorOccurred = true
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:54]: [Suggestion] I think we need to keep the flag, why was it removed?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Looks good. Please find my comment about the flag.
The changes require the test of all scenarios:
 - capture images
 - import images
 - import pdf
 - open with images
 - open with pdf
Could you please create one PR with 2 changes?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift (hunk 1 above)
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:54]: The flag was the reason why the error screen was not triggered.

**Comment by zladzeyka** [comment]
The current version has 2 problems:
1. No internet:
https://user-images.githubusercontent.com/3911816/210357923-ee7f7048-13ea-4f36-ba4a-60c8c1634d84.MP4

2. Internet switched off on the Review screen
https://user-images.githubusercontent.com/3911816/210357957-abd5969a-1ee0-4c9b-8392-7a4dc17feec9.MP4

**Review by zladzeyka — APPROVED** [review]
Looks good 🎉👏🏻

**Pull request merged**